src/HorusStudio.Maui.MaterialDesignControls/Controls/MaterialProgressIndicator.cs

```diff
@@ -8,153 +7,325 @@ public enum MaterialProgressIndicatorType
+                    // So we abort the linear or circular animation here
+                    if (Type == MaterialProgressIndicatorType.Circular
+                        && _customActivityIndicator != null)
                     {
-                        // Handle linear animation
-                        if (IsVisible)
-                            StartLinearAnimation();
-                        else
-                            this.AbortAnimation("LinearAnimation" + Id);
+                        _customActivityIndicator.IsRunning = false;
                     }
-                    break;
-                case "Renderer":
-                    if (!_rendered)
-                        _rendered = true;
-                    else
+                    else if (Type == MaterialProgressIndicatorType.Linear)
                     {
-                        // This property is setted on the view appearing and in the view dissapearing
-                        // So we abort the linear animation here
-                        if (Type == MaterialProgressIndicatorType.Linear)
-                            this.AbortAnimation("LinearAnimation" + Id);
+                        this.AbortAnimation(LinearAnimationName + Id);
                     }
-                    break;
+                }
             }
         }
 
-        public void SetProgressIndicatorType()
+        public void UpdateLayoutAfterTypeChanged(MaterialProgressIndicatorType type)
         {
-            switch (Type)
+            switch (type)
             {
                 case MaterialProgressIndicatorType.Linear:
-                    HeightRequest = 4;
-                    WidthRequest = -1;
-                    _progressBar = new BoxView()
+                    _progressBar = new BoxView
                     {
                         Color = IndicatorColor,
                         IsEnabled = this.IsEnabled,
                         Margin = new Thickness(0),
                     };
-                    this.Content = _progressBar;
-                    this.BackgroundColor = TrackColor;
+                    Content = _progressBar;
+                    BackgroundColor = TrackColor;
                     StartLinearAnimation();
                     break;
                 case MaterialProgressIndicatorType.Circular:
-                    HeightRequest = 48;
-                    WidthRequest = 48;
-                    this.BackgroundColor = Colors.Transparent;
-                    _customActivityIndicator = new CustomActivityIndicator()
+                    BackgroundColor = Colors.Transparent;
+                    _customActivityIndicator = new CustomActivityIndicator
                     {
-                        Color = IndicatorColor,
-                        IsRunning = true
+                        IndicatorColor = IndicatorColor,
+                        TrackColor = Colors.Transparent,
+                        Thickness = 4
```

Comment:
should extract these hardcoded `4` to a const so it can be tracked and modified more easily if needed